### ISSUE TITLE
Not able to execute on Xcode 10 beta 

### ISSUE THREAD

**Issue by badrinathvm** [google/EarlGrey#733]
Seeing below errors while executing tests on Xcode 10 beta.

unexpected duplicate creator 'PBXCp [/Users/enqa/jenkins_slave/workspace/XXX_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/EarlGrey.framework /Users/enqa/jenkins_slave/workspace/XXX_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/Evernote.app/EarlGrey.framework' for node

 ❌  error: Multiple commands produce '<PbxCp /Users/enqa/jenkins_slave/workspace/XXX_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework>':

13:53:48 ❌  error: Multiple commands produce '<CodeSign /Users/enqa/jenkins_slave/workspace/Evernote_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework>':
13:53:48 
13:53:48 
13:53:48 
13:53:48 ❌  error: Multiple commands produce '/Users/enqa/jenkins_slave/workspace/XXX_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework/_CodeSignature':
13:53:48 
13:53:48 
13:53:48 
13:53:48 ❌  error: Multiple commands produce '/Users/enqa/jenkins_slave/workspace/XXX_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework/EarlGrey':
13:53:48 
13:53:48 
13:53:48 
13:53:48 ❌  error: Multiple commands produce '/Users/enqa/jenkins_slave/workspace/XXX_iOS_Automated_UI_Tests_Branch/DerivedData/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework':](url)

**Comment by tirodkar** [comment]
What command are you using here to copy EG into the app under test? It looks like the Copy Files script we have might not be working correctly. Also, which beta is this?

**Comment by badrinathvm** [comment]
I am testing on `Xcode 10 beta 2`

**Comment by badrinathvm** [comment]
It looks like EarlGrey likes to copy its own things , it updates during build.

**Comment by tirodkar** [comment]
Yeah, EG needs to be injected into the app under test. We use a Copy Files Build phase which is automated with the Ruby gem to do this. 

https://github.com/google/EarlGrey/blob/master/docs/install-and-run.md#step-4-final-test-configuration-

Any update with Beta 3?

**Comment by badrinathvm** [comment]
My copy file is exactly like the same as mentioned in the link. ( https://github.com/google/EarlGrey/blob/master/docs/install-and-run.md#step-4-final-test-configuration- )

Below is my suspect: 

We have two targets each of that does the *Copy Files* , since it is executing on new build system is it trying to copy multiples times and finding duplicates ??

**Comment by tirodkar** [comment]
Hmm, I'm not so certain of that, unless if the destination path has been changed. I didn't receive anything like this on beta 3. Is the behavior the same?

**Comment by badrinathvm** [comment]
Each of the targets tried to do a `copy Files` phase. Was able to solve the code signing issue by disabling `code Sign on  copy`  check box.

still facing two more issues


Showing All Errors Only


:-1: Multiple commands produce '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework':

1) Target 'XXXTests' has copy command from '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/EarlGrey.framework' to '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework'
2) Target 'YYYTests' has copy command from '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/EarlGrey.framework' to '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework'
3) Target 'ZZZTests' has copy command from '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/EarlGrey.framework' to '/Users/bvenkatnarayansetty/Library/Developer/Xcode/DerivedData/Lightning-bqwglibfxszbzcbfqlsqwrftorbm/Build/Products/Debug-iphonesimulator/XXX.app/EarlGrey.framework'

**Comment by badrinathvm** [comment]
It's getting copied in all the three targets.
Let me try the options which u mentioned couple of mins..

**Comment by comonitos** [comment]
Go to Xcode->File->Workspace/Project Settings-> Build System -> Legacy Build System.

**Comment by badrinathvm** [comment]
This works on Legacy Build system. but we want to get it build on New build system.

**Comment by badrinathvm** [comment]
Probably having it one of the scheme should work i believe.. But again it might fail because we don't execution order of the build , as this is completely paralell.

**Comment by tirodkar** [comment]
Are you not able to determine the order of parallelism? You could check that in your script possibly?

**Comment by tirodkar** [comment]
It would really help us debug this if you could provide us with a sample project to repro this.

**Comment by badrinathvm** [comment]
Sure, will do..

**Comment by biluncloud** [comment]
Hi, is this problem solved? I met this too.

**Comment by tirodkar** [comment]
This should be fixed at HEAD. We updated the Xcode version. Are you still seeing this?

**Comment by biluncloud** [comment]
> This should be fixed at HEAD. We updated the Xcode version. Are you still seeing this?

We still have this problem, but it's not on this project. The test target in our own project met this. The only thing we can do is changing to the legacy build system. We are using the latest Xcode version. Could you please share how this was fixed? Thanks.

**Comment by tirodkar** [comment]
@biluncloud PTAL @ https://github.com/google/EarlGrey/pull/754

**Comment by biluncloud** [comment]
@tirodkar Thanks, I've already tried this. Not awesome but it works. Hope apple would solve this problem.
